tests/pamiq_core/test_state_persistence.py

```diff
@@ -0,0 +1,111 @@
+import shutil
+import unittest
+from datetime import datetime, tzinfo
+from pathlib import Path
+
+import pytest
+
+from pamiq_core.state_persistence import PersistentStateMixin, StateStore
+
+
+class TestStateStore:
+    state_1 = PersistentStateMixin()
+    state_2 = PersistentStateMixin()
+
+    def test_register(self, tmp_path):
+        store = StateStore(states_dir=tmp_path)
+        store.register("state_1", self.state_1)
+
+        assert store._registered_states == {"state_1": self.state_1}
+
+        store.register("state_2", self.state_2)
+
+        assert store._registered_states == {
+            "state_1": self.state_1,
+            "state_2": self.state_2,
+        }
+
+    def test_register_name_already_used_error(self, tmp_path):
+        store = StateStore(states_dir=tmp_path)
+        store.register("same_name", self.state_1)
+
+        # should raise KeyError:
+        with pytest.raises(KeyError):
+            store.register("same_name", self.state_2)
+
+    def test_save_state(self, tmp_path, mocker):
+        # prepare mock objects
+        parent_mock = mocker.Mock()
+        mock_state_1 = mocker.Mock(spec=PersistentStateMixin)
+        mock_state_2 = mocker.Mock(spec=PersistentStateMixin)
+        parent_mock.attach_mock(mock_state_1, "mock_state_1")
+        parent_mock.attach_mock(mock_state_2, "mock_state_2")
+
+        # configure StateStore object
+        store = StateStore(states_dir=tmp_path)
+        store.register("mock_state_1", mock_state_1)
+        store.register("mock_state_2", mock_state_2)
+
+        # Mock store.datetime.now so that tests do not depend on the current time
+        fixed_test_time = datetime(2025, 2, 27, 12, 0, 0)
+
+        # Cannot directly mock `datetime.now` because it is a built-in method
+        # Instead, create a subclass of `datetime` and mock `now` method
+        class FakeDatetime(datetime):
+            @classmethod
+            def now(cls, tz: tzinfo | None = None):
+                return cls(
+                    year=fixed_test_time.year,
+                    month=fixed_test_time.month,
+                    day=fixed_test_time.day,
+                    hour=fixed_test_time.hour,
+                    minute=fixed_test_time.minute,
+                    second=fixed_test_time.second,
+                    microsecond=fixed_test_time.microsecond,
+                    tzinfo=fixed_test_time.tzinfo,
+                )
+
+        mocker.patch("pamiq_core.state_persistence.datetime", FakeDatetime)
+
+        state_path = store.save_state()
+
+        assert state_path.exists()  # test: folder is created
+        assert state_path == Path(tmp_path / "2025-02-27_12-00-00,000000.state")
+        assert parent_mock.mock_calls == [
+            mocker.call.mock_state_1.save_state(state_path / "mock_state_1"),
+            mocker.call.mock_state_2.save_state(state_path / "mock_state_2"),
+        ]  # test: `save_state()` is called for each registered state
```

Comment:
呼び出し確認はこのようにすると良いと思います。
```suggestion
        mock_state_1.save_state.assert_called_once_with(state_path / "mock_state_1")
        mock_state_2.save_state.assert_called_once_with(state_path / "mock_state_2")
```

Reply:
確かに！
この方が良いですね、ありがとうございます！！！